torch/csrc/distributed/c10d/comm.h

```diff
@@ -31,6 +31,10 @@ class GradBucket {
     return tensors_;
   }
 
+  std::vector<at::Tensor>* getTensorsRef() {
+    return &tensors_;
+  }
```

Comment:
getTensors is not used in C++ code, but there is a python binding to `get_tensors` of `dist._GradBucket`, and it is used in default_hooks.py.

I still want to keep the const semantics for that method, so added this one.

Reply:
In that case can we change the return type to `std::vector<at::Tensor>&` instead of `std::vector<at::Tensor>*`? That is usually the convention we use.